harness/determined/core/_context.py

```diff
@@ -101,7 +101,15 @@ def _dummy_init(
 
     storage_manager = None
     if checkpoint_storage is not None:
-        storage_manager = storage.from_string(checkpoint_storage)
+        if isinstance(checkpoint_storage, str):
+            storage_manager = storage.from_string(checkpoint_storage)
+        elif isinstance(checkpoint_storage, dict):
+            if checkpoint_storage["type"] == "shared_fs":
+                raise ValueError(
+                    "Cannot configure a shared_fs checkpoint storage with a "
+                    "dictionary. Use a string or a configuration file."
+                )
+            storage_manager = det.common.storage.build(checkpoint_storage, container_path=None)
 
```

Comment:
We are validating user input, but I see that we will silently ignore any `checkpoint_storage` parameter which is neither string nor dict.

Can you add an else clause to the `if instance(...)` checks?